acceptance/gossip_peerings_test.go

```diff
@@ -97,11 +102,61 @@ func TestGossipPeerings(t *testing.T) {
 	}
 	defer cluster.Stop()
 
-	done := make(chan struct{})
+	done := make(chan error)
+	checkGossipPeerings(t, cluster, 20, done)
+
+	var err error
+	select {
+	case <-stopper:
+		return
+	case err = <-done:
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// Restart the first node. We're using this for the gossip address.
+	// This shows:
+	// - bootstraping (using ourselves as gossip host)
+	// - joining gossip by being on the receiver side.
+	pickedNode := 0
+	log.Infof("Restarting node %d", pickedNode)
+	if err := cluster.Nodes[pickedNode].Restart(5); err != nil {
+		t.Fatalf("problem restarting node %d: %v", pickedNode, err)
+	}
+	checkGossipPeerings(t, cluster, 20, done)
```

Comment:
Ugh, I just changed the way this worked in an unsent PR so that `checkGossipPeerings` did not run a separate goroutine. This is going to need refactoring when we merge. I think it might be better to make this a completely separate test.